src/main/java/org/reaktivity/command/log/internal/LogQueueDepthCommand.java

```diff
@@ -0,0 +1,104 @@
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package org.reaktivity.command.log.internal;
+
+import org.agrona.LangUtil;
+import org.reaktivity.command.log.internal.layouts.StreamsLayout;
+import org.reaktivity.command.log.internal.spy.RingBufferSpy;
+import org.reaktivity.nukleus.Configuration;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.stream.Stream;
+
+public final class LogQueueDepthCommand
+{
+    private final Path directory;
+    private final boolean verbose;
+    private final Logger out;
+
+    private final long streamsCapacity;
+    private final long throttleCapacity;
+
+    public LogQueueDepthCommand(
+            Configuration config,
+            Logger out,
+            boolean verbose)
+    {
+        this.directory = config.directory();
+        this.out = out;
+        this.verbose = verbose;
+        this.streamsCapacity = config.streamsBufferCapacity();
+        this.throttleCapacity = config.throttleBufferCapacity();
+    }
+
+    private boolean isStreamsFile(
+            Path path)
+    {
+        return path.getNameCount() - directory.getNameCount() == 3 &&
+                "streams".equals(path.getName(path.getNameCount() - 2).toString()) &&
+                Files.isRegularFile(path);
+    }
+
+    private void onDiscovered(
+            Path path)
+    {
+        if (verbose)
+        {
+            out.printf("Discovered: %s\n", path);
+        }
+    }
+
+    private void displayQueueDepth(
+            Path path)
+    {
+        try (StreamsLayout layout = new StreamsLayout.Builder()
+                .path(path)
+                .streamsCapacity(streamsCapacity)
+                .throttleCapacity(throttleCapacity)
+                .readonly(true)
+                .build();
+        )
+        {
+            RingBufferSpy streamsBuffer = layout.streamsBuffer();
+            RingBufferSpy throttleBuffer = layout.throttleBuffer();
+
+            out.printf("streamsReadPointer=%d streamsWritePointer=%d streamsQueueDepth=%d  ",
+                    streamsBuffer.consumerPosition(),
+                    streamsBuffer.producerPosition(),
+                    streamsBuffer.producerPosition() - streamsBuffer.consumerPosition());
+            out.printf("throttleReadPointer=%d throttleWritePointer=%d throttleQueue=%d \n ",
+                    throttleBuffer.consumerPosition(),
+                    throttleBuffer.producerPosition(),
+                    throttleBuffer.producerPosition() - throttleBuffer.consumerPosition());
+        }
```

Comment:
Excellent.

Now need to handle the case where the `producerPosition()` and `consumerPosition()` change on successive calls, so must pull out into local variables rather than re-calling the same methods.

Suggest the following:
```
String owner = path.getName(path.getNameCount() - 1);
displayQueueDepth(String.format("%s.streams", owner), layout.streamsBuffer());
displayQueueDepth(String.format("%s.throttle", owner), layout.throttleBuffer());
```

```
private void displayQueueDepth(
    String name,
    RingBufferSpy buffer)
{
    // read consumer position first for pessimistic queue depth
    long consumerAt = buffer.consumerPosition();
    long producerAt = buffer.producerPosition();

    out.printf("%s=%d %d %d\n ", producerAt - consumerAt, producerAt, consumerAt);
}
```

Reply:
Hello. I applied your suggestion  but I left the printed values this way:
```
out.printf("%s: queueDepth=%d writePointer=%d readPointer=%d\n ",name, producerAt - consumerAt, producerAt, consumerAt);
```
I think it's easier to read the output values.

Reply:
This is noisier than necessary, and the requirement is actually to only print the queue depth.

Please change it to:
```
out.printf("%s: %d", name, producerAt - consumerAt);
```